openupgradelib/openupgrade.py

```diff
@@ -2606,19 +2606,90 @@ def delete_record_translations(cr, module, xml_ids):
         ),
     )
     for row in cr.fetchall():
-        query = """
-            DELETE FROM ir_translation
-            WHERE module = %s AND name LIKE %s AND res_id = %s;
-        """
-        logged_query(
-            cr,
-            query,
-            (
-                module,
-                row[0] + ",%",
-                row[1],
-            ),
-        )
+        model = row[0]
+        record_id = row[1]
+        if version_info[0] < 16:
+            query = """
+                DELETE FROM ir_translation
+                WHERE module = %s AND name LIKE %s AND res_id = %s;
+            """
+            logged_query(
+                cr,
+                query,
+                (
+                    module,
+                    model + ",%",
+                    record_id,
+                ),
+            )
+        else:
+            from psycopg2.extras import Json
+
+            table = model.replace(".", "_")
+            cr.execute(
+                """
+                SELECT isc.column_name
+                FROM information_schema.columns isc
+                JOIN ir_model_fields imf ON (
+                    imf.name = isc.column_name AND imf.model = %s)
+                JOIN ir_model_data imd ON (
+                    imd.model = 'ir.model.fields' AND imd.res_id = imf.id)
+                WHERE isc.table_name = % AND imd.module = %s
+                    AND imf.translate""",
+                (model, table, module),
+            )
+            list_columns = [x[0] for x in cr.fetchall()]
+            if not list_columns:
+                continue
+            columns = ", ".join(list_columns)
+            query = """
+                SELECT {columns}
+                FROM {table}
+                WHERE id = %s""".format(
+                table=table,
+                columns=columns,
+            )
+            cr.execute(
+                query,
+                (record_id,),
+            )
+            list_translations = [x[0] for x in cr.fetchall()]
+            new_translations = []
+            for column, translations in zip(list_columns, list_translations):
+                field_translations = translations or {}
+                if "en_US" in field_translations:
+                    clean_value = {"en_US": field_translations["en_US"]}
+                else:
+                    clean_value = {
+                        "en_US": (
+                            column[:-4]
```

Comment:
Don't understand this. Please add comments.

Reply:
This part was taken from Odoo:

![Selection_1472](https://github.com/OCA/openupgradelib/assets/25005517/f1584852-5d76-4842-b4e0-944808c98d18)

Reply:
And comment added.

Reply:
I just deleted this part, as it was not making sense:

![Selection_1473](https://github.com/OCA/openupgradelib/assets/25005517/fe3b0bb3-fa1f-406f-92cc-88af0f6c8222)

With my code, that empty cells would have been filled with `{"en_US": "Description"}`